lms/djangoapps/grades/tasks.py

```diff
@@ -115,6 +115,28 @@ def compute_grades_for_course(course_key, offset, batch_size, **kwargs):  # pyli
             raise result.error
 
 
+@task(
+    bind=True,
+    time_limit=SUBSECTION_GRADE_TIMEOUT_SECONDS,
+    max_retries=2,
+    default_retry_delay=RETRY_DELAY_SECONDS,
+    routing_key=settings.RECALCULATE_GRADES_ROUTING_KEY
+)
+def recalculate_grades_course_and_subsection(self, **kwargs):
+    user = kwargs.get('user')
+    course_key = kwargs.get('course_key')
+
+    if not user or course_key:
+        log.error('Failed to start grade recalculation for Course and Subsection, Missing user or course_key info.')
+
+    previous_course_grade = CourseGradeFactory().read(user, course_key=course_key)
+    if previous_course_grade and previous_course_grade.attempted:
+        CourseGradeFactory().update(
```

Comment:
This `update()` method takes an optional `collected_block_structure` argument, which defaults to None.  `CourseGradeFactory.read()` returns a `CourseGrade` object, which has a `course_data` attribute, which in turn has a `collected_structure` attribute.  This could be passed into the `update()` call, and would save us an extra course block structure lookup (I think).

From my devstack LMS:
```python
user = User.objects.get(username='staff')
key = CourseKey.from_string('course-v1:edX+DemoX+Demo_Course')
grade = CourseGradeFactory().read(user, course_key=key)
collected_structure = grade.course_data.collected_structure
CourseGradeFactory().update(
    user=user,
    course_key=key,
    force_update_subsections=True,
    collected_block_structure=collected_structure
)
```